### PR TITLE
Improvements in Victory.py

### PR DESCRIPTION
**Pull request by MeuhV1**
Removed double function
Discover first x era technology goals fail if target is not met and all techs from that era are discovered
Discover first x era technology goals only checks when the discovered era is the era for the UHV
Tamil trade UHV stores decimal values
Fixed Enter first era goals

### CODE REVIEW

File: Assets/Python/Victory.py [hunk 1]
```diff
@@ -2811,6 +2811,16 @@ def countFirstDiscovered(iPlayer, iEra):
 			iCount += 1
 	return iCount
 	
+def isFirstDiscoveredPossible(iPlayer, iEra, iRequired):
```
Comment (dguenms) [inline Assets/Python/Victory.py:2877]: Wouldn't be better to just count techs first discovered by `iPlayer` and all undiscovered techs `countFirstDiscovered(-1, iEra)` and see if they add up to at least `iRequired`?

**Review by dguenms — CHANGES_REQUESTED** [review]

File: Assets/Python/Victory.py (hunk 1 above)
Comment (MeuhV1) [reply Assets/Python/Victory.py:2877]: Yeah. I'll change it.

**Comment by MeuhV1** [comment]
Updated

File: Assets/Python/Victory.py [hunk 2]
```diff
@@ -1599,18 +1599,24 @@ def onTechAcquired(iPlayer, iTech):
...
 				else: lose(iJapan, 2)
+			if not isFirstDiscoveredPossible(iJapan iGlobal, 8) or not isFirstDiscoveredPossible(iJapan, iDigital, 8):
+				lose(iJapan, 2)
 				
 		# third English goal: be the first to discover ten Renaissance and ten Industrial technologies
 		if isPossible(iEngland, 2):
 			if countFirstDiscovered(iPlayer, iRenaissance) >= 8 and countFirstDiscovered(iPlayer, iIndustrial) >= 8:
 				if iPlayer == iEngland: win(iEngland, 2)
 				else: lose(iEngland, 2)
+			if not isFirstDiscoveredPossible(iEngland iRenaissance, 8) or not isFirstDiscoveredPossible(iEngland, iIndustrial, 8):
```
Comment (dguenms) [inline Assets/Python/Victory.py]: Looks like there is a comma missing here for all these invocations.

**Review by dguenms — CHANGES_REQUESTED** [review]

**Comment by MeuhV1** [comment]
Updated.

I also included an additional condition so the UHV checks only if the discovered tech belongs to the era of the tech goals for speed improvement.

**Comment by MeuhV1** [comment]
Rebased to latest DoC
Included Tamil UHV update
Fixed Enter first era goals

File: Assets/Python/Victory.py [hunk 3]
```diff
@@ -532,21 +532,21 @@ def checkTurn(iGameTurn, iPlayer):
 				
 		# third goal: acquire 4000 gold by trade by 1200 AD
 		if isPossible(iTamils, 2):
-			iTradeGold = 0
+			iTradeGold = 0.0
```
Comment (dguenms) [inline Assets/Python/Victory.py]: Instead of a float, this should still store an integer but instead by a factor of 100, Civ4 does not need more accuracy.

**Review by dguenms — CHANGES_REQUESTED** [review]

File: Assets/Python/Victory.py (hunk 3 above)
Comment (MeuhV1) [reply Assets/Python/Victory.py]: Only iTradeGold of all instances of fTamilTradeGold?

Comment (dguenms) [reply Assets/Python/Victory.py]: All instances where floats are used to store the accrued Tamil gold.

Comment (MeuhV1) [reply Assets/Python/Victory.py]: Done.

I also included Italy to the list of conditional spawns and moved the code inside the check of line 1129.

**Comment by MeuhV1** [comment]
Used int * 100 for Tamil UHV

I also included Italy to the list of conditional spawns and moved the code inside the check of line 1129 of RNF.py

**Review by dguenms — APPROVED** [review]

**Pull request merged**